src/ghost-content/index.js

```diff
@@ -0,0 +1,113 @@
+import path from 'path'
+import fs from 'fs'
+import Promise from 'bluebird'
+import glob from 'glob'
+import uuid from 'uuid'
+import get from 'lodash/get'
+
+Promise.promisifyAll(fs)
+const globAsync = Promise.promisify(glob)
+
+module.exports = ({ logger, db, projectLocation }) => {
+  logger.info('Ghost Content Manager initialized')
+
+  const upsert = ({ knex, tableName, where, data, idField = 'id' }) =>
+    knex(tableName)
+      .where(where)
+      .select(idField)
+      .then(res => {
+        const id = get(res, '0.id')
+        return id
+          ? knex(tableName)
+              .where(idField, id)
+              .update(data)
+              .then()
+          : knex(tableName)
+              .insert(Object.assign({}, where, data))
+              .then()
+      })
+
+  const recordFile = async (folderPath, folder, file) => {
+    const knex = await db.get()
+    const filePath = path.join(folderPath, file)
+    await fs.readFileAsync(filePath, 'utf-8').then(content =>
+      upsert({
+        knex,
+        tableName: 'ghost_content',
+        where: { folder, file },
+        data: { content }
+      })
+    )
+  }
+
+  const normalizeFolder = folder => {
+    const folderPath = path.resolve(projectLocation, folder)
+    return {
+      folderPath,
+      folder: path.relative(projectLocation, folderPath)
+    }
+  }
+
+  const addFolder = async (folder, filesGlob) => {
+    const f = normalizeFolder(folder)
+    const files = await globAsync(filesGlob, { cwd: f.folderPath })
+    await Promise.map(files, file => recordFile(f.folderPath, f.folder, file))
+  }
+
+  const recordRevision = async (folder, file, content) => {
+    const knex = await db.get()
+
+    const f = normalizeFolder(folder)
+    folder = f.folder
```

Comment:
@emirotin , I wouldn't recommend mutating input params (this may sometimes lead to unexpected behavior).

Reply:
I don't usually do that but it's OK in a short method.
Not sure though what you mean by unexpected :) All changes are obviously local to the function scope so changing the pointer is safe (unlike in languages that support `out` params). On contrary of course it's strictly discouraged to mutate the _properties_ of passed params, like `opts.x = y`